rocketpy/Flight.py

```diff
@@ -2161,6 +2161,41 @@ def acceleration(self):
         """Rocket acceleration magnitude as a rocketpy.Function of time."""
         return (self.ax**2 + self.ay**2 + self.az**2) ** 0.5
 
+    @cached_property
+    def max_acceleration_power_on(self):
+        """Maximum acceleration reached by the rocket during motor burn."""
+        closest_value = float("inf")
+        index_to_slice = None
+
+        for index, item in enumerate(self.acceleration):
+            if abs(item[0] - self.rocket.motor.burn_out_time) < abs(
+                closest_value - self.rocket.motor.burn_out_time
+            ):
+                closest_value = item[0]
+                index_to_slice = index
+        max_acceleration_time_index = np.argmax(
+            self.acceleration[: index_to_slice + 1, 1]
+        )
```

Comment:
The `for` loop seems to be searching the index of the acceleration time array that is the closest to the burn out time. Since the time array is ordered, wouldn't it be much faster to perform a binary search as follows:

```py
from bisect import bisect_left

def find_closest(ordered_sequence, value):
    pivot_index = bisect_left(ordered_sequence, value)
    if pivot_index == 0:
        return pivot_index
    if pivot_index == len(ordered_sequence):
        return pivot_index - 1
    
    smaller, greater = ordered_sequence[pivot_index-1], ordered_sequence[pivot_index]
    
    return pivot_index - 1 if value - smaller <= greater - value else pivot_index
```

If I understood the problem correctly, this function could be placed in `tools.py` and has performance $\mathcal{O}\left(\log(n)\right)$.